docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6328,409 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+as a table or multiple tables. A table here is any data structure with one or
+more entries and each entry can be represented as a set of key fields and data
+fields.  Actions in P4 tables behave like unions and that has been borrowed here
+in `GenericTable`.  This can be potentially used targets to map to either TDI
+based targets, or even non-TDI based targets. If using TDI, then potentially,
+for every new feature to be added, only TDI target support and remote client
+code will be required to make use of that function.  All other middle layers
+should theoretically remain unchanged.
+
+P4Runtime supports inserting, modifying, deleting and reading `GenericTable`
+entries with the `GenericTa bleEntry` entity, which has the following fields:
+
+* `table_id`, which identifies the table instance; the `table_id` is determined
+  by the P4Info message.
+
+* `match`, a repeated field of `GenericFieldMatch` messages. Each element in the
+  repeated field is used to provide a value for the corresponding element in the
+  P4info.
+
+* `union`, which indicates which of the table's unions to execute in case of
+  match and with which argument values.
+
+* `priority`, a 32-bit integer used to order entries when the table's match key
+  includes an optional, ternary or range match.
+
+* `is_default_entry`, a boolean flag which indicates whether the table entry is
+  the default entry for the table. See [Default entry](#sec-default-entry)
+  section for more information.
+
+* `metadata`, an arbitrary `bytes` value which is opaque to the
+  target. There is no requirement of where this is stored, but it must be
+  returned by the server along with the rest of the entry when the client
+  performs a read on the entry.
+
+The `priority` field must be set to a non-zero value if the match key includes a
+ternary match (&ie; in the case of PSA if the P4Info entry for the table
+indicates that one or more of its match fields has an `OPTIONAL`, `TERNARY` or
+`RANGE` match type) or to zero otherwise. A higher priority number indicates
+that the entry must be given higher priority when performing a table lookup.
+Clients must allow multiple entries to be added with the same priority value.
+If a packet can match multiple entries with the same priority, it is not
+deterministic in the data plane which entry a packet will match.  If a client
+wishes to make the matching behavior deterministic, it must use different
+priority values for any pair of table entries that the same packet matches.
+
+The `match` and `priority` fields are used to uniquely identify an entry within
+a table. Therefore, these fields cannot be modified after the entry has been
+inserted and must be provided for `MODIFY` and `DELETE` updates. When deleting
+an entry, these key fields (along with `is_default_action`) are the only fields
+considered by the server. All other fields must be ignored, even if they have
+nonsensical values (such as an invalid action field). In the case of a *keyless*
```

Comment:
This looks like a copy & paste bug: there is no "action" field in GenericTableEntry.  Replace with `union`?